day2_exercises/into4D.py

```diff
@@ -0,0 +1,113 @@
+#%% Into four dimensions
+# -*- coding: utf-8 -*-
+"""
+Created on Sat Nov  7 13:46:24 2015
+Exercise for Fall 2015
+@author: Elissaios
+"""
+# Compatibility with Python 3
+from __future__ import print_function, division
+
+# Standard imports of libraries
+import numpy as np
+import nibabel as nib
+import matplotlib.pyplot as plt
+
+# Declarations
+TR = 2.5
+std_vol = []
+ave_task = []
+ave_rest = []
+outlier_task = []
+outlier_rest = []
+
+# Input data
+folder = r"C:\Users\Elissaios\Box Sync\Projects\fMRI\analysis1";
+filePath = folder + r"\ds114_sub009_t2r1.nii";
+condPath = folder + r"\ds114_sub009_t2r1_cond.txt";
+img = nib.load(filePath)
+data = img.get_data()
+#print(img.shape)
+task = np.loadtxt(condPath)
+task[:,0:2] = task[:,0:2]/TR  # change sec to TR for first two columns (2 is for 1+1)
+dur = list(img.shape)[3]
+nTR = dur
+time_course = np.zeros(nTR)
+nBlock = task.shape[0]
+
+for itr in range(0,nBlock):
+    durBlock = task[itr,1]
+    blockS = int(task[itr,0])                # block start
+    blockE = int(blockS+durBlock)     # block end
+    time_course[blockS:blockE] = 1
+
+plt.figure(0)
+plt.ylim(-0.5,1.5)
+plt.plot(time_course)
+
+# Create two matrices, one for each condition; it includes first 10 s of rest
+is_task_tr = [time_course==1]
```

Comment:
You don't need the list brackets around the variable.  The extra brackets are why you need squeeze and ``np.array`` in the lines below.   Simplify with:

```
is_task_tr = time_course==1
data_task = data[..., is_task_tr]
```